PULL REQUEST - Actividad1

herrerameri opened the pull request:
**Entrega de actividad 1**
¡Inicio de TODOMAN! 

![todoman](https://cloud.githubusercontent.com/assets/22835487/19741549/64d42700-9bc1-11e6-9830-0afdfa49d187.jpg)
![captura_celular](https://cloud.githubusercontent.com/assets/22835487/19765876/11bea97a-9c4b-11e6-9285-34060b1d9298.png)
![captura_tablet](https://cloud.githubusercontent.com/assets/22835487/19765875/11b564c8-9c4b-11e6-828c-b61354df98fe.png)

herrerameri committed:
- Manejo de fragments: creación de 2 fragments que son utilizados de manera diferente en dispositivos con menor y mayor tamaño de pantalla.
- DatePickerFragment para selección de una fecha.
- Ícono de la aplicación

herrerameri committed:
- Restructuración del proyecto
- Merge de repositorio de cristiantanas

herrerameri committed:
Merge branch 'master' of https://github.com/cristiantanas/TODOman

# Conflicts:
#	todoman/src/main/AndroidManifest.xml
#	todoman/src/main/java/org/uab/dedam/todoman/HomeActivity.java
#	todoman/src/main/java/org/uab/dedam/todoman/NewTaskActivity.java
#	todoman/src/main/res/layout/activity_home.xml
#	todoman/src/main/res/layout/activity_newtask.xml
#	todoman/src/main/res/values/strings.xml

cristiantanas commented:
¡Muy buen trabajo! Sólo algunos comentarios:
- En el método `onCreateView` de los Fragments deberíamos devolver simplemente la vista. Cualquier otra acción debería ir en algunos de los otros métodos, como por ejemplo `onActivityCreated`.
- Relacionado con esto, en la clase _FragmentNewTask_ habría que asegurarse que la Activity realmente implementa la interface _DatePickerFragment.OnDateSetCallback_:

``` Java
((DatePickerFragment.OnDateSetCallback)getActivity()).selectDate();
```

Seria mejor definir una interfaz similar a como se implementa _OnDateSetCallback_ del _DatePickerFragment_ y siempre que hay que comunicar alguna cosa a la Activity, la comunicación pasaría a través de esta interface.
- Habría que controlar siempre que las Activities implementen las interfaces que definimos:

``` Java
public void onAttach(Activity activity) {
     super.onAttach(activity);
     try {
          this.listener = (OnDateSetCallback) activity;
     } catch(ClassCastException exception) {
          throw new ClassCastException("The activity does not implement the interface.");
     }
}
```
- En la HomeActivity no acabo de entender porque la doble comprobación `existsFrgNewTask`. Si llegamos al evento `onClick` del _FloatingActionButton_ será porque el fragment no existe, no?

herrerameri committed:
Correcciones PR#1
- Catchear la posible excepción de ClassCastException para Activities que interactúan con DatePickerFragment
- Quitar código que no sea retorno de layout en onCreateView de fragmentos
- Corrijo error condicional duplicado en HomeActivity.java
- Corrijo a inglés "Lista de tareas" por "TASKS"
